test/e2e/test/elasticsearch/checks_keystore.go

```diff
@@ -50,14 +58,6 @@ func CheckESKeystoreEntries(k *test.K8sClient, b Builder, expectedKeys []string)
 				return err
 			}
 
-			// rolling upgrade leading to that correct keystore configuration should eventually be over,
-			// with cluster health green
-			if err := allPodsReady(b, k); err != nil {
-				return err
-			}
-			if err := clusterHealthGreen(b, k); err != nil {
-				return err
-			}
 			return nil
```

Comment:
Turns out @thbkrkr was right and I was wrong in https://github.com/elastic/cloud-on-k8s/pull/1794#discussion_r328193291. Reversing the order here makes more sense, and does not prevent the test from checking the right thing. It's possible we pass the first 2 blocks (rolling upgrade over) before the rolling upgrade is even started, but then we still fail the keystore check anyway. So that's fine.

Reply:
:)